compiler/rustc_borrowck/src/diagnostics/mod.rs

```diff
@@ -623,6 +623,28 @@ impl UseSpans<'_> {
         }
     }
 
+    // callback a subdiag on the label-span
+    pub(super) fn var_span_subdiag_label(
+        self,
+        err: &mut Diagnostic,
+        cb: impl Fn(Span) -> crate::session_diagnostics::CaptureVarCause,
```

Comment:
>`impl Trait` not allowed in `Fn` trait return.

I was thinking this case shouldn't be that general anyway since the function is specific to `captuared var cause`